modules/21_tax/on/equations.gms

```diff
@@ -105,6 +106,14 @@ v21_taxrevNetNegEmi(t,regi) =e= cm_frac_NetNegEmi * pm_taxCO2eqSum(t,regi) * v21
 q21_emiAllco2neg(t,regi)..
 v21_emiALLco2neg(t,regi) =e= -vm_emiAll(t,regi,"co2") + v21_emiALLco2neg_slack(t,regi);
 
+***---------------------------------------------------------------------------
+*'  Calculation of PE tax: tax rate times primary energy
+*'  Documentation of overall tax approach is above at q21_taxrev.
+***---------------------------------------------------------------------------
+q21_taxrevPE(t,regi)$(t.val ge max(2010,cm_startyear))..
+v21_taxrevPE(t,regi) =e= p21_tau_pe_tax(t,regi,"pegas") * vm_prodPe(t,regi,"pegas")
+                          - p21_taxrevPE0(t,regi);
+
```

Comment:
That should be formulated generally for all PEs. That way, other people can use the tax for other PE energy carriers and purposes. I will dm you about it.